src/annotator/test/frame-observer-test.js

```diff
@@ -140,10 +140,16 @@ describe('annotator/frame-observer', () => {
     it("doesn't trigger onFrameAdded when annotatable iframe is from a different domain", async () => {
       const iframe = createAnnotatableIFrame();
       iframe.setAttribute('src', 'http://cross-origin.dummy');
+      let expectedError;
 
-      await onDocumentReady(iframe);
+      try {
+        await onDocumentReady(iframe);
+      } catch (error) {
+        expectedError = error;
```

Comment:
So the fact that `onDocumentReady` resolved previously with a cross-origin iframe was a bug?

Also, can you clarify why the `onDocumentReady` call is needed here? The other tests just use the `await waitForFrameObserver()` step before checking whether the frame was injected or not.

Reply:
> So the fact that onDocumentReady resolved previously with a cross-origin iframe was a bug?

No, it wasn't. The cross-origin iframe fired the `load` event normally. The expected error would raised later, when trying to execute the `_onFrameAdded` function to inject the client. However, that line is wrapped in a `try/catch` and we tested the `console.warn` execution. With the changes in this PR, however, the error is thrown in the `onDocumentReady`.

> Also, can you clarify why the onDocumentReady call is needed here? 

It is not strictly necessary. I added `onDocumentReady` because the iframe may not load  in 50s (`waitForFrameObserver`). It may take a few more milliseconds more to do to try to fetch a non-existent URL.

I have remove it anyway because it doesn't seem necessary.

Reply:
Actually, I will put it back because sometimes it does take a bit longer than 50s:

```
Chrome Headless 93.0.4577.0 (Linux x86_64) annotator/frame-observer FrameObserver doesn't trigger onFrameAdded when annotatable iframe is from a different domain FAILED
	AssertError: expected warn to be called once but was called 0 times
	    at Function.fail (/home/runner/work/client/client/node_modules/sinon/pkg/sinon.js:174:25)
	    at failAssertion (/home/runner/work/client/client/node_modules/sinon/pkg/sinon.js:120:20)
	    at Function.assert.<computed> [as calledOnce] (/home/runner/work/client/client/node_modules/sinon/pkg/sinon.js:149:17)
	    at Context.<anonymous> (/tmp/annotator/test/frame-observer-test.js:147:14 <- /tmp/31ce315758c0f34b60d4b4aa2b02be64.browserify.js:222901:14)
Chrome Headless 93.0.4577.0 (Linux x86_64): Executed 456 of 2894 (1 FAILED) (0 secs / 1.279 secs)
WARN: 'Unable to inject the Hypothesis client (from 'http://localhost:9876/context.html' into a cross-origin frame 'http://cross-origin.dummy/')'
```

You can see above that the `console.warn` function has been executed (from the catch block) but the after the test has finished.

In this example the CI in Jenkins finished OK, while in GH didn't (with the above error):
![Screenshot 2021-09-16 at 10 45 49](https://user-images.githubusercontent.com/8555781/133580943-de320069-50f7-4660-af24-891e86e1a6b1.png)

Reply:
> It may take a few more milliseconds more to do to try to fetch a non-existent URL.

How about using an invalid local URL instead (eg. `http://localhost:$PORT` where $PORT is some value we can be decently sure is not likely to be in use)?